lib/modules/manager/cargo/extract.ts

```diff
@@ -286,9 +289,45 @@ export async function extractPackageFile(
     'Cargo.lock',
   );
   const res: PackageFileContent = { deps };
-  // istanbul ignore if
   if (lockFileName) {
+    logger.debug(
+      `Found lock file ${lockFileName} for packageFile: ${packageFile}`,
+    );
+
+    const versionsByPackage = await extractLockFileVersions(lockFileName);
+    if (!versionsByPackage) {
+      logger.debug(
+        `Could not extract lock file versions from ${lockFileName}.`,
+      );
+      return res;
+    }
+
     res.lockFiles = [lockFileName];
+
+    for (const dep of deps) {
+      const packageName = dep.packageName ?? dep.depName!;
+      const versions = coerceArray(versionsByPackage.get(packageName));
+      try {
+        const lockedVersion = versioning.getSatisfyingVersion(
+          versions,
+          dep.currentValue!,
+        );
+        if (lockedVersion) {
+          dep.lockedVersion = lockedVersion;
+        } else {
+          logger.debug(
+            `No locked version found for package ${dep.depName} in the range of ${dep.currentValue}.`,
+          );
+        }
+      } catch (err) {
+        // istanbul ignore next: this should not happen
```

Comment:
can this be triggered by a invalid version inside the toml file?

Reply:
@viceice No. If I have an invalid version inside the toml file, the code will enter the `else` branch at L318, and no `lockedVersion` will not be found in `res?.deps[0]`.

Reply:
i think we should remove the try-catch then and let throw any unexpected error to get more visibillity

Reply:
Done.